git_machete/client.py

```diff
@@ -602,8 +602,14 @@ def connected_with_green_edge(bd: LocalBranchShortName) -> bool:
             else:
                 return
 
-        ans = self.ask_if(f"\nBranch {bold(down_branch)} is now merged into {bold(branch)}. Slide {bold(down_branch)} out of the tree of branch dependencies?" + get_pretty_choices('y', 'N'),
-                          f"\nBranch {bold(down_branch)} is now merged into {bold(branch)}. Sliding {bold(down_branch)} out of the tree of branch dependencies...", opt_yes=opt_yes)
+        remote = self.__git.get_strict_remote_for_fetching_of_branch(branch)
+        ans = self.ask_if(f"\nBranch {bold(branch)} is now fast-forwarded to match {bold(down_branch)}. Push {bold(branch)} to {bold(remote)}?" + get_pretty_choices('y', 'N'),
+                          f"\nBranch {bold(branch)} is now fast-forwarded to match {bold(down_branch)}. Pushing {bold(branch)} to {bold(remote)}...", opt_yes=opt_yes)
+        if ans in ('y', 'yes'):
+            self.__git.push(remote, branch)
+
+        ans = self.ask_if(f"\nBranch {bold(branch)} is now pushed to {bold(remote)}. Slide {bold(down_branch)} out of the tree of branch dependencies?" + get_pretty_choices('y', 'N'),
```

Comment:
Hmmm this doesn't need to be true, what if the user chose `N` to `Push...`?

Reply:
I'd leave it as it was, with `...is now merged...`

Reply:
@PawelLipski I added new logic to change the msg depending on the answer.